deps/build.jl

```diff
@@ -1,37 +1,47 @@
-using BinaryProvider # requires BinaryProvider 0.3.0 or later
+gmsh_root = nothing
 
-# Parse some basic command-line arguments
-const verbose = "--verbose" in ARGS
-const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
-products = [
-    LibraryProduct(prefix, ["libgmsh"], :libgmsh),
-]
-
-# Download binaries from hosted location
-bin_prefix = "https://github.com/ahojukka5/GmshBuilder/releases/download/v4.4.1"
-
-# Listing of files generated by BinaryBuilder:
-download_info = Dict(
-    MacOS(:x86_64) => ("$bin_prefix/gmsh.v4.4.1.x86_64-apple-darwin14.tar.gz", "a29d761a6826bbca16e500b195c1ddd22487fd62ae8be99930e3755a1e170c33"),
-    Linux(:x86_64, libc=:glibc) => ("$bin_prefix/gmsh.v4.4.1.x86_64-linux-gnu.tar.gz", "f9bb150e70877ad1c2da1edae697f5275776ffcff9c682f666d58f032c87ae89"),
-)
-
-# Install unsatisfied or updated dependencies:
-unsatisfied = any(!satisfied(p; verbose=verbose) for p in products)
-dl_info = choose_download(download_info, platform_key_abi())
-if dl_info === nothing && unsatisfied
-    # If we don't have a compatible .tar.gz to download, complain.
-    # Alternatively, you could attempt to install from a separate provider,
-    # build from source or something even more ambitious here.
-    error("Your platform (\"$(Sys.MACHINE)\", parsed as \"$(triplet(platform_key_abi()))\") is not supported by this package!")
+if haskey(ENV,"GMSHROOT") && !isempty(ENV["GMSHROOT"])
+  gmsh_root = ENV["GMSHROOT"]
+  @info """
+  Using the gmsh installation found via the GMSHROOT environment variable.
+  GMSHROOT=$gmsh_root"""
+else
+  using gmsh_jll
+  GMSH_FOUND = gmsh_jll.is_available()
+  if GMSH_FOUND
+    gmsh_root = gmsh_jll.artifact_dir
+    @info """
+    Using the gmsh installation installed via BinaryBuilder."""
+  end
 end
 
-# If we have a download, and we are unsatisfied (or the version we're
-# trying to install is not itself installed) then load it up!
-if unsatisfied || !isinstalled(dl_info...; prefix=prefix)
-    # Download and install binaries
-    install(dl_info...; prefix=prefix, force=true, verbose=verbose)
+GMSH_FOUND = false
+if gmsh_root != nothing
+  gmsh_bin = (Sys.iswindows() ? joinpath(gmsh_root,"bin","gmsh.exe") : 
+              joinpath(gmsh_root,"bin","gmsh"))
+  if isfile(gmsh_bin)
+    GMSH_FOUND = true
+    @info "gmsh binary found in $gmsh_bin"
+  else
+    @warn """
+    gmsh binary not found in $gmsh_bin
+    Make sure that GMSHROOT points to a correct gmsh installation"""
+  end
+  gmsh_jl = joinpath(gmsh_root,"lib","gmsh.jl")
+  if isfile(gmsh_jl)
+    GMSH_FOUND = true
+    @info "Using the gmsh Julia API found in $gmsh_jl"
+  else
+    @warn """
+    gmsh Julia API not found in $gmsh_jl
+    Make sure that you have installed the full gmsh SDK"""
+  end
 end
 
-# Write out a deps.jl file that will contain mappings for our products
-write_deps_file(joinpath(@__DIR__, "deps.jl"), products, verbose=verbose)
+open("deps.jl","w") do f
```

Comment:
what is the purpose of this? Is this just for debugging stuff?

Reply:
ok nvm just found it, is there are a reason why this is still written to a file? I actually would prefer to just have the variables `GMSH_FOUND` and `gmsh_jl` inside `src/Gmsh.jl`